ygot/render.go

```diff
@@ -1231,15 +1219,33 @@ func unionInterfaceValue(v reflect.Value, appendModuleName bool) (interface{}, e
 	case v.Elem().Elem().NumField() != 1:
 		return nil, fmt.Errorf("received a union type which did not have one field, had: %v", v.Elem().Elem().NumField())
 	}
```

Comment:
Consider creating a helper IsInterfaceToStructPtr in utils. 
Also, first two cases imply you can have a ptr to ptr to struct - is this expected?

Reply:
No - good catch. Not sure where the first test for ptr came from. :-)

Reply:
Created `IsValueInterfaceToStructPtr` in the helper library and added a bunch of unit tests for the functionality therein.